### PR TITLE
[Feature] Create Usage Records for Usage Based Billing

### PR DESCRIPTION
**Pull request by lizkenyon**
Allow users to create ussage records for their usage based plans
* If a developer passes in a subscription line item ID, we will attempt to make the record on that subscription line it
* Otherwise we will query the subscriptions for the active useage billing
* We allow passing in an idempotency key if the developer would like. This is forwarded on to the API request

### WHY are these changes introduced?

Fixes #0000 

## Type of change

- [ ] Patch: Bug (non-breaking change which fixes an issue)
- [x] Minor: New feature (non-breaking change which adds functionality)
- [ ] Major: Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have used `pnpm changeset` to create a draft changelog entry (do NOT update the `CHANGELOG.md` files manually)
- [x] I have added/updated tests for this change
- [x] I have documented new APIs/updated the documentation for modified APIs (for public APIs)

### CODE REVIEW

File: packages/apps/shopify-api/lib/billing/check.ts [hunk 1]
```diff
@@ -133,6 +133,7 @@ const HAS_PAYMENTS_QUERY = `
         id
         name
         test
+        status
```
Comment (lizkenyon) [inline packages/apps/shopify-api/lib/billing/check.ts:136]: We were not returning the status from this method. So we had no way to know which subscription was actually active.

File: packages/apps/shopify-api/lib/billing/__tests__/create-usage-record.test.ts [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+    const response = await shopify.billing.createUsageRecord({
+      session,
+      description,
+      price,
+      subscriptionLineItemId: subscriptionId,
+    });
+
+    expect(response).toEqual(
+      JSON.parse(Responses.USAGE_RECORD_CREATE_RESPONSE).data
+        .appUsageRecordCreate.appUsageRecord,
```
Comment (paulomarg) [inline packages/apps/shopify-api/lib/billing/__tests__/create-usage-record.test.ts:63]: Nit: if we'll be using this response to test more than a couple of tests, we could consider exporting it as a separate const to make these `expect` calls a bit easier to read.

File: packages/apps/shopify-api/lib/billing/create-usage-record.ts [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+
+interface CreateUsageRecordVariables {
+  description: string;
+  price: Money;
+  subscriptionLineItemId: string;
+  idempotencyKey?: string;
+}
+
+const CREATE_USAGE_RECORD_MUTATION = `
+mutation appUsageRecordCreate($description: String!, $price: MoneyInput!, $subscriptionLineItemId: ID!) {
```
Comment (paulomarg) [inline packages/apps/shopify-api/lib/billing/create-usage-record.ts:28]: Should we name this in a more "unique" way for tracking purposes? Could be a follow-up if we're not in other methods.

File: packages/apps/shopify-app-remix/src/server/authenticate/admin/billing/create-usage-record.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+    try {
+      return await api.billing.createUsageRecord({
+        session,
+        description: options.description,
+        price: options.price,
+        isTest: options.isTest,
+        idempotencyKey: options.idempotencyKey,
+        subscriptionLineItemId: options.subscriptionLineItemId,
+      });
```
Comment (paulomarg) [inline packages/apps/shopify-app-remix/src/server/authenticate/admin/billing/create-usage-record.ts:27]: Any particular reason not to pass the options through?

```suggestion
      return await api.billing.createUsageRecord({
        ...options,
        session,
      });
```

File: packages/apps/shopify-api/docs/guides/billing.md [hunk 5]
```diff
@@ -284,4 +284,16 @@ const canceledSubscription = await shopify.billing.cancel({
...
+| ---------------------  | ---------------------------- | :-------: | :-----------: | ---------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `session`              | `Session`                    |    Yes    |       -       | The session to be used for this request                                                                                                                                          |
+| `description`          | `string`                     |    Yes    |       -       | The description of the usage record.                                                                                                                  |
+| `price`                | `Money`                      |    Yes    |       -       | The amount and currency to be charged                                                                                              |
```
[the remark this answers is not in the document]
Comment (lizkenyon) [reply packages/apps/shopify-api/docs/guides/billing.md:294]: Yeah I can update the documentation, to clearly specify amount and currency.

> to pass the objects we use for the billing API in general through
With this are you suggesting we don't use the Money type, in our types?

[the remark this answers is not in the document]
Comment (paulomarg) [reply packages/apps/shopify-api/docs/guides/billing.md:294]: I think we use it in responses, but not in any of the params for the methods, so this one stands out a bit.

**Pull request merged**